cdap-data-fabric/src/main/java/co/cask/cdap/data2/transaction/Transactions.java

```diff
@@ -49,12 +49,26 @@ public static void abortQuietly(TransactionSystemClient txClient, Transaction tx
 
   public static void invalidateQuietly(TransactionSystemClient txClient, Transaction tx) {
     try {
-      txClient.invalidate(tx.getWritePointer());
+      if (!txClient.invalidate(tx.getWritePointer())) {
+        LOG.error("Failed to invalidate transaction {}", tx);
+      }
     } catch (Throwable t) {
       LOG.error("Exception when invalidating transaction {}", tx, t);
     }
   }
 
+  /**
+   * Wraps the given {@link Throwable} as a {@link TransactionFailureException} if it is not already an instance of
+   * {@link TransactionFailureException}.
+   */
+  public static TransactionFailureException asTransactionFailure(Throwable t) {
```

Comment:
The generic message does not any meaning. See further below for an example of where the old code added more context to the message. I am not sure whether Tephra's messages will always tell you what went wrong - did commit() fail? Or invalidate()? The stack trace would reveal that but that will require reading code to understand where it happened.